### PR TITLE
Allow users to access their own domain portal without app permission

### PR DESCRIPTION
**Pull request by selfhoster1312**
Fix for #2384

Removed some indentation levels and added some comments for clarity.

EDIT: tested

### CODE REVIEW

File: src/authenticators/ldap_ynhuser.py [hunk 1]
```diff
@@ -54,11 +58,12 @@ def user_is_allowed_on_domain(user: str, domain: str) -> bool:
...
-        else:
-            parent_domain = domain.split(".", 1)[-1]
-            return user_is_allowed_on_domain(user, parent_domain)
-
+        parent_domain = domain.split(".", 1)[-1]
+        return user_is_allowed_on_domain(user, parent_domain)
+    
     mtime = portal_settings_path.stat().st_mtime
+    # TODO: Why would that stored mtime EVER NOT be < time.time()??
+    # What is this condition supposed to check/prevent?
```
Comment (selfhoster1312) [inline src/authenticators/ldap_ynhuser.py:66]: I'd really like to have an explanation for this :)

**Review by alexAubin — APPROVED** [review]
LGTM unless you want to test if and/or fix the weird `if` mtime thing

**Comment by selfhoster1312** [comment]
Ok i fixed a typo and tested this is all good on my side.

> LGTM unless you want to test if and/or fix the weird if mtime thing

Well i don't understand what it's supposed to do / check ? But i don't understand how a time set in the past could be further into the future than now's time :)

What is the entire condition supposed to check/prevent?

**Comment by alexAubin** [comment]
>What is the entire condition supposed to check/prevent?

It's supposed to cache the relevant info from `{domain}.json` instead of reading it every time. It's generated by `app_ssowatconf()` so it's not likely to change very often

**Comment by selfhoster1312** [comment]
Good for me.

**Pull request merged**